Purchases/Public/RCPurchases.m

```diff
@@ -386,7 +386,7 @@ - (void)identify:(NSString *)appUserID completionBlock:(nullable RCReceivePurcha
             if (error == nil) {
                 [self updateCachesWithCompletionBlock:completion];
             } else {
-                CALL_IF_SET(completion, nil, error);
```

Comment:
maybe to make this less error-prone, we could rename these macros to 
`CALL_ON_SAME_THREAD` and `CALL_ON_MAIN_THREAD`, so it's very obvious to devs that they should decide on a thread

Reply:
I believe we should be calling dispatch here because it's possible that `identifyAppUserID` makes a thread change if the user is anonymous. The problem is that if it doesn't make the thread change (because the user is not anoymous), we would be dispatching unnecessarily.

Reply:
I think we need consistency. It should always come back on background, or always on main, but shouldn't switch depending on internals.

From my experience, APIs that always callback on the main thread are easier to use and trip up people less often.

If you want to avoid the pointless dispatch, you can check if it's on the main thread and call directly, else dispatch.

Reply:
We should probably just have one single macro that checks if it's in main thread and dispatches to the main thread if not. That way we would be consistent and we will avoid unnecessary dispatches. 

Is it really a problem if we dispatch to the main thread from the main thread? We can probably just do it without impact.

Reply:
that's what `CALL_AND_DISPATCH_IF_SET` does, I believe
https://github.com/RevenueCat/purchases-ios/blob/master/Purchases/Public/RCPurchases.m#L935

I think we should:
- rename the methods so their meaning is abundantly obvious (see my suggestion above)
- always call them, I agree with Jacob that returning on the main thread is somewhat standard for good SDKs and it's also easier on our side to have that set as the standard. 
- potentially rename `dispatch` to `dispatch_on_main_thread` or even better `execute_on_main_thread`, since it won't dispatch if you're already on the main thread. 
- `CALL_IF_SET` might not be needed at all, although I'm not opposed to keeping it

Reply:
> Is it really a problem if we dispatch to the main thread from the main thread?

Not really. There is a slight benefit to calling straight into the block since it keeps the stack trace cleaner and might help debugging but that's it.

Reply:
to clarify, `CALL_AND_DISPATCH_IF_SET` checks if you're on the main thread and just calls straight away, and dispatches otherwise. So you're good from a debugging perspective.

Reply:
Totally right, the implementation of dispatch checks for if being called from main thread already. I will make the changes we have talked about.